### PR TITLE
authentication: Refactor OIDC provider middleware and handler creation

### PR DESCRIPTION
**Pull request by brancz**
@observatorium/maintainers

### CODE REVIEW

File: authentication/oidc.go [hunk 1]
```diff
@@ -83,10 +87,24 @@ func NewOIDC(logger log.Logger, configs []TenantOIDCConfig) (http.Handler, map[s
...
+	return fmt.Sprintf("observatorium_%s", tenant)
+}
+
+type OIDCProvider struct {
+	logger           log.Logger
+	provider         *oidc.Provider
+	client           *http.Client
+	config           OIDCConfig
+	cookieName       string
+	callbackRedirect string
```
Comment (squat) [inline authentication/oidc.go:100]: callback and redirect are usually synonyms in OAuth terminology. We should rather call this either "callbackURL" or "redirectURL"

**Review by squat — COMMENTED** [review]
overall looks great :))
If we can add godoc comments for the exported methods and types that would be enough for me :)))

**Comment by brancz** [comment]
Done :)

File: authentication/oidc.go (hunk 1 above)
Comment (brancz) [reply authentication/oidc.go]: Agreed. Renamed to `redirectURL`.

**Review by squat — APPROVED** [review]
lgtm

**Pull request merged**